src/unix/os390.c

```diff
@@ -996,3 +1081,44 @@ int uv__io_fork(uv_loop_t* loop) {
   uv__platform_loop_delete(loop);
   return uv__platform_loop_init(loop);
 }
+
+
+int uv__os390_connect(uv_connect_t* req, uv_stream_t* handle,
+                      const struct sockaddr* addr, unsigned int addrlen) {
+  struct aiocb* aio_connect;
+  uv__io_t* w;
+  uv__os390_epoll* ep;
+  int rv;
+  int rc;
+  int rsn;
+ 
+  w = &handle->io_watcher;
+  ep = handle->loop->ep;
+  aio_connect = &req->aio;
+  memset(aio_connect, 0, sizeof(*aio_connect));
+  aio_connect->aio_fildes = w->fd;
+  aio_connect->aio_notifytype = AIO_MSGQ;
+  aio_connect->aio_cmd = AIO_CONNECT;
+  aio_connect->aio_msgev_qid = ep->msg_queue;
+  aio_connect->aio_sockaddrlen = addrlen;
+  aio_connect->aio_sockaddrptr = (struct sockaddr_in*)uv__malloc(addrlen);
+  if (aio_connect->aio_sockaddrptr == NULL)
+    return -1;
+
+  /* The BPX4AIO call requires the socket to be in blocking mode. */
+  rv = uv__nonblock(w->fd, 0);
+  if (rv) {
```

Comment:
`uv__free(aio_connect->aio_sockaddrptr); aio_connect->aio_sockaddrptr = NULL;`?  Or is it eventually freed elsewhere?

Reply:
It is freed on line 815 (after we get the response from the message queue).

Reply:
Okay, I thought it might.  Can you add an explaining comment?  It won't be obvious to the casual reader.

Would it make sense to change it to a union or a `struct sockaddr_storage`?  The less you have to worry about memory management, the better, IMO.

Reply:
Done, used sockaddr_storage instead.